alteryx/featuretools
# v1.0.0

[GUIDE: Transitioning to Featuretools Version 1.0](https://featuretools.alteryx.com/en/latest/resources/transition_to_ft_v1.0.html)

**v1.0.0 Oct 12, 2021**

* Enhancements
    * Add support for creating EntitySets from Woodwork DataTables (#1277)
    * Add ``EntitySet.__deepcopy__`` that retains Woodwork typing information (#1465)
    * Add ``EntitySet.__getstate__`` and ``EntitySet.__setstate__`` to preserve typing when pickling (#1581)
    * Returned feature matrix has woodwork typing information (#1664)
* Fixes
    * Fix ``DFSTransformer`` Documentation for Featuretools 1.0 (#1605)
    * Fix ``calculate_feature_matrix`` time type check and ``encode_features`` for synthesis tests (#1580)
    * Revert reordering of categories in ``Equal`` and ``NotEqual`` primitives (#1640)
    * Fix bug in ``EntitySet.add_relationship`` that caused ``foreign_key`` tag to be lost (#1675)
    * Update DFS to not build features on last time index columns in dataframes (#1695)
* Changes
    * Remove ``add_interesting_values`` from ``Entity`` (#1269)
    * Move ``set_secondary_time_index`` method from ``Entity`` to ``EntitySet`` (#1280)
    * Refactor Relationship creation process (#1370)
    * Replaced ``Entity.update_data`` with ``EntitySet.update_dataframe`` (#1398)
    * Move validation check for uniform time index to ``EntitySet`` (#1400)
    * Replace ``Entity`` objects in ``EntitySet`` with Woodwork dataframes (#1405)
    * Refactor ``EntitySet.plot`` to work with Woodwork dataframes (#1468)
    * Move ``last_time_index`` to be a column on the DataFrame (#1456)
    * Update serialization/deserialization to work with Woodwork (#1452)
    * Refactor ``EntitySet.query_by_values`` to work with Woodwork dataframes (#1467)
    * Replace ``list_variable_types`` with ``list_logical_types`` (#1477)
    * Allow deep EntitySet equality check (#1480)
    * Update ``EntitySet.concat`` to work with Woodwork DataFrames (#1490)
    * Add function to list semantic tags (#1486)
    * Initialize Woodwork on feature matrix in ``remove_highly_correlated_features`` if necessary (#1618)
    * Remove categorical-encoding as an add-on library (will be added back later) (#1632)
    * Remove autonormalize as an add-on library (will be added back later) (#1636)
    * Remove tsfresh, nlp_primitives, sklearn_transformer as an add-on library (will be added back later) (#1638)
    * Update input and return types for ``CumCount`` primitive (#1651)
    * Standardize imports of Woodwork (#1526)
    * Rename target entity to target dataframe (#1506)
    * Replace ``entity_from_dataframe`` with ``add_dataframe`` (#1504)
    * Create features from Woodwork columns (#1582)
    * Move default variable description logic to ``generate_description`` (#1403)
    * Update Woodwork to version 0.4.0 with ``LogicalType.transform`` and LogicalType instances (#1451)
    * Update Woodwork to version 0.4.1 with Ordinal order values and whitespace serialization fix (#1478)
    * Use ``ColumnSchema`` for primitive input and return types (#1411)
    * Update features to use Woodwork and remove ``Entity`` and ``Variable`` classes (#1501)
    * Re-add ``make_index`` functionality to EntitySet (#1507)
    * Use ``ColumnSchema`` in DFS primitive matching (#1523)
    * Updates from Featuretools v0.26.0 (#1539)
    * Leverage Woodwork better in ``add_interesting_values`` (#1550)
    * Update ``calculate_feature_matrix`` to use Woodwork (#1533)
    * Update Woodwork to version 0.6.0 with changed categorical inference (#1597)
    * Update ``nlp-primitives`` requirement for Featuretools 1.0 (#1609)
    * Remove remaining references to ``Entity`` and ``Variable`` in code (#1612)
    * Update Woodwork to version 0.7.1 with changed initialization (#1648)
    * Removes outdated workaround code related to a since-resolved pandas issue (#1677)
    * Remove unused ``_dataframes_equal`` and ``camel_to_snake`` functions (#1683)
    * Update Woodwork to version 0.8.0 for improved performance (#1689)
    * Remove redundant typecasting in ``encode_features`` (#1694)
    * Speed up ``encode_features`` if not inplace, some space cost (#1699)
    * Clean up comments and commented out code (#1701)
    * Update Woodwork to version 0.8.1 for improved performance (#1702)
* Documentation Changes
    * Add a Woodwork Typing in Featuretools guide (#1589)
    * Add a resource guide for transitioning to Featuretools 1.0 (#1627)
    * Update ``using_entitysets`` page to use Woodwork (#1532)
    * Update FAQ page to use Woodwork integration (#1649)
    * Update DFS page to be Jupyter notebook and use Woodwork integration (#1557)
    * Update Feature Primitives page to be Jupyter notebook and use Woodwork integration (#1556)
    * Update Handling Time page to be Jupyter notebook and use Woodwork integration (#1552)
    * Update Advanced Custom Primitives page to be Jupyter notebook and use Woodwork integration (#1587)
    * Update Deployment page to use Woodwork integration (#1588)
    * Update Using Dask EntitySets page to be Jupyter notebook and use Woodwork integration (#1590)
    * Update Specifying Primitive Options page to be Jupyter notebook and use Woodwork integration (#1593)
    * Update API Reference to match Featuretools 1.0 API (#1600)
    * Update Index page to be Jupyter notebook and use Woodwork integration (#1602)
    * Update Feature Descriptions page to be Jupyter notebook and use Woodwork integration (#1603)
    * Update Using Koalas EntitySets page to be Jupyter notebook and use Woodwork integration (#1604)
    * Update Glossary to use Woodwork integration (#1608)
    * Update Tuning DFS page to be Jupyter notebook and use Woodwork integration (#1610)
    * Fix small formatting issues in Documentation (#1607)
    * Remove Variables page and more references to variables (#1629)
    * Update Feature Selection page to use Woodwork integration (#1618)
    * Update Improving Performance page to be Jupyter notebook and use Woodwork integration (#1591)
    * Fix typos in transition guide (#1672)
    * Update installation instructions for 1.0.0rc1 announcement in docs (#1707`, :pr:`1708`, :pr:`1713`, :pr:`1716)
    * Fixed broken link for Demo notebook in README.md (#1728)
    * Update ``contributing.md`` to improve instructions for external contributors (#1723)
    * Manually revert changes made by :pr:`1677` and :pr:`1679`.  The related bug in pandas still exists. (#1731)
* Testing Changes
    * Remove entity tests (#1521)
    * Fix broken ``EntitySet`` tests (#1548)
    * Fix broken primitive tests (#1568)
    * Added Jupyter notebook cleaner to the linters (#1719)
    * Update reviewers for minimum and latest dependency checkers (#1715)
    * Full coverage for EntitySet.__eq__ method (#1725)
    * Add tests to verify all primitives can be initialized without parameter values (#1726)
